src/components/navigation/LinkBase.js

```diff
@@ -0,0 +1,30 @@
+import { downcastRef } from '../../util/typing';
+
+/**
+ * @typedef {import('../../types').BaseProps} BaseProps
+ * @typedef {import('preact').JSX.HTMLAttributes<HTMLAnchorElement>} HTMLAnchorAttributes
+ */
+
+/**
+ * Base component for Link components. Applies common attributes.
+ *
+ * @param {BaseProps & HTMLAnchorAttributes} props
+ */
+export default function LinkBase({
+  children,
+  className,
+  elementRef,
+
+  ...htmlAttributes
+}) {
+  return (
+    <a
+      {...htmlAttributes}
```

Comment:
Right: it's not the rule that the HTML attributes rest props will _always_ come first, but it will probably be the case frequently. I think it should be the rough default case, as it avoids ambiguity of precedence for any component-controlled attributes. And exceptions would be documented or commented. Does that sound about right?